spec/models/service_spec.rb

```diff
@@ -732,6 +732,42 @@
     end
   end
 
+  describe '#add_to_service' do
+    let(:service) { FactoryGirl.create(:service) }
+    let(:child_service) { FactoryGirl.create(:service) }
+
+    it 'associates a child_service to the service' do
+      child_service.add_to_service(service)
+
+      expect(service.reload.services).to include(child_service)
+    end
+
+    it 'raise an error if the child_service is already part of a service' do
+      child_service.add_to_service(service)
+
+      expect { child_service.add_to_service(service) }.to raise_error MiqException::Error
+    end
+  end
+
+  describe '#remove_from_service' do
+    let(:service) { FactoryGirl.create(:service) }
+    let(:child_service) { FactoryGirl.create(:service) }
+
+    it 'associates a child_service to the service' do
+      child_service.add_to_service(service)
+      expect(child_service.service).to eq(service)
+
+      child_service.remove_from_service(service)
+      expect(child_service.service).to be_nil
+    end
+
+    it 'raise an error if the child_service is already part of a service' do
+      child_service.add_to_service(service)
+
+      expect { child_service.add_to_service(service) }.to raise_error MiqException::Error
+    end
+  end
+
```

Comment:
@gmcculloug Can we add a spec for circular reference check

Reply:
There are existing tests for circular references in the service model [here](https://github.com/ManageIQ/manageiq/blob/master/spec/models/service_spec.rb#L263) and the service template model starting [here](https://github.com/ManageIQ/manageiq/blob/master/spec/models/service_template_spec.rb#L243).